### ISSUE TITLE
custom components(trainer, loss, label, augmentation...)

### ISSUE THREAD

**Issue by develooper1994** [awslabs/djl#137]
## Question
I should say, i am impressed. i was thinking about like that for C++ and Python. You guys getting me start to java-language :].
How can i make a custom components? I have some custom layers in my model that process audio.

**Comment by frankfliu** [comment]
@develooper1994 
Thank you for looking into DJL. We don't have documentation about how to implement a custom Layer yet. Here is a brief idea:
- DJL use Block (Layer in Keras) to build a neuron network. You can extends[ AbstractBlock](https://javadoc.io/doc/ai.djl/api/latest/ai/djl/nn/AbstractBlock.html) to create your own Block
- You can find some built-in Block implementation: https://github.com/awslabs/djl/blob/master/api/src/main/java/ai/djl/nn/recurrent/LSTM.java
- In most case, you don't need override Trainer, you can use trainer to implement your own training loop
- You can create your own Loss class or use SimpleCompositeLoss to combine existing Loss

Please join our [Slack channel](https://join.slack.com/t/deepjavalibrary/shared_invite/zt-ar91gjkz-qbXhr1l~LFGEIEeGBibT7w), we can better answer your questions.

**Comment by develooper1994** [comment]
I don't know java language (yet). But read your documentation and understand some ideas but **djl documentation has to be extended.**
If Pytorch is at the document, tutorials level, it becomes popular.
